src/OpenTelemetry/Logs/ILogger/OpenTelemetryLogger.cs

```diff
@@ -153,9 +152,18 @@ internal static void SetLogRecordSeverityFields(ref LogRecordData logRecordData,
         LogRecord logRecord,
         ref LogRecord.LogRecordILoggerData iLoggerData,
         in TState state,
+        bool includeAttributes,
         bool parseStateValues)
     {
-        iLoggerData.State = null;
+        if (!includeAttributes)
```

Comment:
Add check for `state == null` here and remove the `else if` condition [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4609/files#diff-48886d84cf3065379188763f0948382a1b89bf9ebf9bdef6f8146fa6acf4776eR198-R200).
```suggestion
        if (!includeAttributes || state is null)
```

Reply:
I think the else if could only be removed if we also take your suggestion:

> How about we also remove the support for parsing custom typed objects as well in 1.5.1? This PR https://github.com/open-telemetry/opentelemetry-dotnet/pull/4560 is doing that anyway?

Right?

Reply:
My thinking here was most states will not be `null` so better to not spend cycles checking for uncommon case until it is needed. But I moved it up and rewrote the check so that the JIT should elide it for structs completely so there shouldn't be any impact.